### ISSUE TITLE
Probably wrong instruction to install jitsi from debian repository

### ISSUE THREAD

**Issue by mabra** [jitsi/jitsi#705]
*This Issue tracker is only for reporting bugs and tracking code related issues.*

Before posting, please make sure you check community.jitsi.org to see if the same or similar bugs have already been discussed. General questions, installation help, and feature requests can also be posted to community.jitsi.org.

## Description
---
Cannot install jitsi via debian repository
## Current behavior
---
After following the instructions given here:
https://desktop.jitsi.org/Main/DebianRepository.html
where all steps are done, an
"apt install jitsi" issues only:
E: Unable to locate package jitsi

## Expected Behavior
---
Installs properly.

## Possible Solution
---
Please update the instructions of fix the repository.

## Steps to reproduce
---
Don't know. Follow the instructions in:
https://desktop.jitsi.org/Main/DebianRepository.html
Searching the repository says, that there is no jitsi desktop version in it!

# Environment details
---
Debian 10, Buster
5.4.0-0.bpo.4-amd64 #1 SMP Debian 5.4.19-1~bpo10+1 (2020-03-09) x86_64 GNU/Linux

**Comment by reini-1** [comment]
Same Problem on Ubuntu 18.04 and 20.04. In the repo is no "jitsi" package available and the Debian package that can be downloaded form the website (https://github.com/jitsi/jitsi/releases/download/Jitsi-2.10/jitsi_2.10.5550-1_amd64.deb) did not work. It freezes on startup with "NimbuzzAvatars".
The same problem as discussed in the closed issue #162 (https://github.com/jitsi/jitsi/issues/162).
It seems jitsi is broken for month on Ubuntu/Debian ...

**Comment by reini-1** [comment]
As mentioned in #675 (https://github.com/jitsi/jitsi/issues/675), jitsi only works with java 8.

**Comment by sebastian-schmid** [comment]
same here the jitsi package ist not in the repo: http://download.jitsi.org/stable/
would be great if installation and update via apt will be made possible

**Comment by karthicraghupathi** [comment]
I'm running into this issue as well. I'm following the instructions from https://desktop.jitsi.org/Main/DebianRepository.html.

It appears others are also experiencing this issue. Here is a post on SuperUser with the same issue: https://superuser.com/q/1576201/82074

Here are the steps I've taken that demonstrates this:

```bash
~$ wget -qO - https://download.jitsi.org/jitsi-key.gpg.key | sudo apt-key add -
[sudo] password for krruser: 
OK

~$ sudo sh -c "echo 'deb https://download.jitsi.org stable/' > /etc/apt/sources.list.d/jitsi-stable.list"

~$ sudo apt update
Hit:1 http://us.archive.ubuntu.com/ubuntu focal InRelease
Get:2 http://us.archive.ubuntu.com/ubuntu focal-security InRelease [109 kB]                                            
Get:3 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]                                            
Hit:4 http://ppa.launchpad.net/system76/pop/ubuntu focal InRelease                                                     
Hit:5 http://packages.microsoft.com/repos/edge stable InRelease                                                        
Hit:6 http://apt.pop-os.org/proprietary focal InRelease                                                                
Get:7 http://us.archive.ubuntu.com/ubuntu focal-updates InRelease [114 kB]                                             
Get:8 https://packages.microsoft.com/repos/vscode stable InRelease [3,959 B]                                           
Hit:10 https://download.jitsi.org stable/ InRelease                                                                    
Ign:11 https://repo.centrify.com/deb stable InRelease                                                                  
Get:12 http://us.archive.ubuntu.com/ubuntu focal-backports InRelease [101 kB]
Get:13 https://repo.centrify.com/deb stable Release [2,709 B]         
Get:14 http://us.archive.ubuntu.com/ubuntu focal-security/main amd64 DEP-11 Metadata [24.3 kB]
Get:15 http://us.archive.ubuntu.com/ubuntu focal-security/universe amd64 DEP-11 Metadata [56.6 kB]    
Hit:9 https://packagecloud.io/slacktechnologies/slack/debian jessie InRelease
Get:16 http://us.archive.ubuntu.com/ubuntu focal-updates/universe Sources [132 kB]
Get:17 http://security.ubuntu.com/ubuntu bionic-security/main amd64 DEP-11 Metadata [49.0 kB]
Get:19 http://us.archive.ubuntu.com/ubuntu focal-updates/main Sources [328 kB]
Get:20 http://us.archive.ubuntu.com/ubuntu focal-updates/main i386 Packages [378 kB]
Get:21 http://us.archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages [670 kB]
Get:22 http://us.archive.ubuntu.com/ubuntu focal-updates/main amd64 DEP-11 Metadata [236 kB]
Get:23 http://us.archive.ubuntu.com/ubuntu focal-updates/universe amd64 Packages [697 kB]
Get:24 http://us.archive.ubuntu.com/ubuntu focal-updates/universe i386 Packages [518 kB]
Get:25 http://us.archive.ubuntu.com/ubuntu focal-updates/universe Translation-en [136 kB]
Get:26 http://us.archive.ubuntu.com/ubuntu focal-updates/universe amd64 DEP-11 Metadata [205 kB]
Get:27 http://us.archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 DEP-11 Metadata [2,468 B]
Get:28 http://us.archive.ubuntu.com/ubuntu focal-backports/universe amd64 DEP-11 Metadata [1,768 B]
Fetched 3,853 kB in 3s (1,102 kB/s)                                            
Reading package lists... Done
Building dependency tree       
Reading state information... Done
All packages are up to date.


~$ sudo apt install jitsi
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package jitsi


~$ cat /etc/apt/sources.list.d/jitsi-stable.list 
deb https://download.jitsi.org stable/


~$ apt search jitsi
Sorting... Done
Full Text Search... Done
jibri/stable 8.0-61-g99288dc-1 all
  Jibri

jicofo/stable 1.0-508-1 i386
  JItsi Meet COnference FOcus

jigasi/stable 1.1-126-g6df3db2-1 amd64
  Jitsi Gateway for SIP

jitsi-archive-keyring/stable 1.0.1 all
  The public key for the Jitsi packages repository

jitsi-meet/stable 2.0.5142-1 all
  WebRTC JavaScript video conferences

jitsi-meet-prosody/stable 1.0.4466-1 all
  Prosody configuration for Jitsi Meet

jitsi-meet-tokens/stable 1.0.4466-1 all
  Prosody token authentication plugin for Jitsi Meet

jitsi-meet-turnserver/stable 1.0.4466-1 all
  Configures coturn to be used with Jitsi Meet

jitsi-meet-web/stable 1.0.4466-1 all
  WebRTC JavaScript video conferences

jitsi-meet-web-config/stable 1.0.4466-1 all
  Configuration for web serving of Jitsi Meet

jitsi-upload-integrations/stable 0.15.15-1 all
  jitsi-upload-integrations

jitsi-videobridge/stable 1126-1 amd64
  WebRTC compatible Selective Forwarding Unit (SFU)

jitsi-videobridge2/stable 2.1-376-g9f12bfe2-1 all
  WebRTC compatible Selective Forwarding Unit (SFU)
```

Should I be using a different package name? I'm looking to install the SIP based VOIP client.

Thanks for your time.

**Comment by freddii** [comment]
> Same Problem on Ubuntu 18.04 and 20.04. In the repo is no "jitsi" package available and the Debian package that can be downloaded form the website (https://github.com/jitsi/jitsi/releases/download/Jitsi-2.10/jitsi_2.10.5550-1_amd64.deb) did not work. It freezes on startup with "NimbuzzAvatars".
> The same problem as discussed in the closed issue #162 (#162).
> It seems jitsi is broken for month on Ubuntu/Debian ...

you also have to install  http://download.jitsi.org/stable/jitsi-archive-keyring_1.0.1_all.deb

**Comment by EureDudeheit** [comment]
I have the same problem on Kubuntu 20.04

@freddii your workaround runs.

**Comment by jose1711** [comment]
Not sure about @freddii's solution since that is a way how to install `jitsi-meet`, not the original `jitsi` SIP (and IRC, Jabber..) client. This is what worked for me on Ubuntu 20.04:
- add Debian repo + update as per instructions at https://desktop.jitsi.org/Main/DebianRepository.html
- download jitsi `.deb` package manually, install via `dpkg -i jitsi_2.10.5550-1_amd64.deb`
- you'll receive errors (but the package will still be installed)
- `apt --fix-broken install` should install missing `jitsi-archive-keyring`
- in order to run `jitsi` you'll need to switch java version to 8:
  ```
  update-alternatives --set java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
  ```

**Comment by KlavsKlavsen** [comment]
@jose1711 On Ubuntu 20.04 atleast - the above fails as its depending on several packages that aren't there:
```
The following packages have unmet dependencies:
 jitsi : Depends: libavformat57 but it is not installable or
                  libavformat-ffmpeg56 but it is not installable
         Depends: libavcodec57 but it is not installable or
                  libavcodec-extra57 but it is not installable or
                  libavcodec-ffmpeg56 but it is not installable or
                  libavcodec-ffmpeg-extra56 but it is not installable
         Depends: libavfilter6 but it is not installable or
                  libavfilter-extra6 but it is not installable or
                  libavfilter-ffmpeg5 but it is not installable
         Depends: libavutil55 but it is not installable or
                  libavutil-ffmpeg54 but it is not installable
         Depends: libswscale4 but it is not installable or
                  libswscale-ffmpeg3 but it is not installable
E: Unable to correct problems, you have held broken packages.
```

I added both stable and unstable jitsi repo (jitsi package is ONLY in unstable).

:(

I also tried the appPackage workaround - but runs jitsi meet.. not jitsi :(

**Comment by traud** [comment]
@KlavsKlavsen when it comes to Ubuntu 20.04 LTS (and Debian 10):
(a) If you do not need H.264 video codec, you [override](http://serverfault.com/a/506993) those [dependencies](https://github.com/jitsi/jitsi/pull/620).
(b) If you need H.264, you have to [compile yourself …](https://askubuntu.com/a/1324208)

**Comment by mabra** [comment]
Another option for the devs would be to convert it to a flatpak packages.

**Comment by ibauersachs** [comment]
@mabra If you want to help with that, please create a PR to the `new-build` branch.

**Comment by sylv-io** [comment]
FYI: Same Issue on Ubuntu 21.04.
The [instructions](https://desktop.jitsi.org/Main/DebianRepository.html) did not work (package `jitsi` not found)

## Workaround:
```bash
# install java 8 (with regards to #675)
$ sudo apt install openjdk-8-jre
# change default java version 
$ sudo update-alternatives --config java
# get deb package
$ wget https://github.com/jitsi/jitsi/releases/download/Jitsi-2.10/jitsi_2.10.5550-1_amd64.deb
# install deb package
$ sudo dpkg -i jitsi_2.10.5550-1_amd64.deb
```

**Comment by Neustradamus** [comment]
Some people has tested on Debian 11?
- https://www.debian.org/News/2021/20210814

The new version, 2 years after Debian 10:
- https://www.debian.org/News/2019/20190706

**Comment by sylv-io** [comment]
> Some people has tested on Debian 11?
> 
> * https://www.debian.org/News/2021/20210814
> 
> The new version, 2 years after Debian 10:
> 
> * https://www.debian.org/News/2019/20190706

I tested it on a clean debian11 VM:
```
sylv@debian11:~$ wget -qO - https://download.jitsi.org/jitsi-key.gpg.key | sudo apt-key add -
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
OK
sylv@debian11:~$  sudo sh -c "echo 'deb https://download.jitsi.org stable/' > /etc/apt/sources.list.d/jitsi-stable.list"
sylv@debian11:~$ sudo apt-get -y update
Hit:1 http://security.debian.org/debian-security bullseye-security InRelease
Hit:2 http://deb.debian.org/debian bullseye InRelease                                                                      
Hit:3 http://deb.debian.org/debian bullseye-updates InRelease                                                              
Hit:4 https://cli.github.com/packages stable InRelease                                           
Get:5 https://download.jitsi.org stable/ InRelease [1,882 B]               
Get:6 https://download.jitsi.org stable/ Packages [54.4 kB]
Fetched 56.2 kB in 2s (33.7 kB/s)   
Reading package lists... Done
sylv@debian11:~$ sudo apt-get -y install jitsi
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package jitsi is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'jitsi' has no installation candidate
```

It is still not working

**Comment by damencho** [comment]
Yep, the package can be found in releases on github

**Comment by sylv-io** [comment]
> Yep, the package can be found in releases on github

Sure, but this issue addresses the missing package in the "debian repository".
If it is not maintained anymore, it shoud be removed from the homepage and documentation .(https://desktop.jitsi.org/Main/DebianRepository.html)

**Comment by sylv-io** [comment]
FIY: looks like debian repository is not supported anymore.

https://github.com/jitsi/jitsi-desktop-site/commit/b052d6a41eebf38e6c2b798c902c3d9dfb1579b1

**Comment by mathieujobin** [comment]
Most recent packages on GitHub are still broken. This is very sad this excellent software went unmaintained. How come they released 2.14.695 after 2.14.721 ?

I thought I had broked it trying to get androidstudio to behave on Kubuntu 23.10
but the fix to use `sudo update-alternatives --config java` with 2.10 suggested by @sylv-io 
still works ! Thank you
